### PR TITLE
Update document model to be product based

### PR DESCRIPTION
**Pull request by sjamcox**
## Purpose of Pull Request
Updated documents to contain all the reviews for a single product, versus having a separate document for each review. This led to great improvements in query speed and overall size of storage. Also added index creation to the seeding script.

### CODE REVIEW

**Comment by sjamcox** [comment]
Postgres commit has been merged to main already.

File: database-couchdb/dataGenerator.js [hunk 1]
```diff
@@ -19,51 +19,64 @@ const generateReviews = async () => {
...
-      productId: faker.random.number({ min: 1, max: 10000000 }),
       color: faker.commerce.color(),
       configuration: faker.lorem.words(),
-      customerId: faker.random.number({ min: 1, max: 10000000 }),
-      customerName: faker.name.findName(),
-      customerCountry: faker.address.country(),
-      profilePicUrl: 'http://placeimg.com/60/60/people',
       category: 'Amazon Devices',
       isBestSeller: faker.random.boolean(),
+      reviews: []
```
Comment (samasno) [inline database-couchdb/dataGenerator.js:31]: looks like a very efficient setup

File: database-couchdb/dataGenerator.js [hunk 2]
```diff
@@ -19,51 +19,64 @@ const generateReviews = async () => {
...
     }
-    documents.push(review);
+    let totalReviews = Math.floor(Math.random() * 10)
+    for (let j = 0; j < totalReviews; j++) {
+      let review = {
+        reviewId: faker.random.uuid(),
```
Comment (samasno) [inline database-couchdb/dataGenerator.js:36]: good that you kept a means of identifying individual reviews

**Review by samasno — APPROVED** [review]
good step up in efficiency

**Pull request merged**